File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/AudioRoom/View/Chat/PanoChatTextCell.m
//
//  PanoChatTextCell.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/19.
//

#import "PanoChatTextCell.h"
#import "PanoDefine.h"
#import <Masonry/Masonry.h>

@implementation PanoChatTextCell {
    UIView *wrapView;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier: reuseIdentifier];
    if (self) {
        wrapView = [UIView new];
        wrapView.layer.cornerRadius = 5;
        wrapView.layer.masksToBounds = true;
        [self.contentView addSubview:wrapView];
        
        self.backgroundColor = [UIColor clearColor];
        self.contentView.backgroundColor = [UIColor clearColor];

        contentLabel = [UILabel new];
        contentLabel.font = fontMedium();
        contentLabel.textColor = [UIColor whiteColor];
        contentLabel.numberOfLines = 0;
        [wrapView addSubview:contentLabel];
        
        [wrapView mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.mas_equalTo(self.contentView).insets(UIEdgeInsetsMake(0, defaultMargin, 0, defaultMargin));
        }];
        
        [contentLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.mas_equalTo(self->wrapView).insets(UIEdgeInsetsMake(8, 8, 8, 8));
        }];
    }
    return self;
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/AudioRoom/View/Chat/PanoChatTextCell.swift
//
//  PanoChatTextCell.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/11/4.
//

import UIKit

class PanoChatTextCell: UITableViewCell {
    
    var contentLabel: UILabel!
    var wrapView: UIView!
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        wrapView = UIView()
        wrapView.layer.cornerRadius = 5
        wrapView.layer.masksToBounds = true
        self.contentView.addSubview(wrapView)
        
        self.backgroundColor = UIColor.clear
        self.contentView.backgroundColor = UIColor.clear
    
        contentLabel = UILabel()
        contentLabel.font = FontMedium
        contentLabel.textColor = UIColor.white
        contentLabel.numberOfLines = 0
        wrapView.addSubview(contentLabel)
        
        wrapView.snp.makeConstraints { (make) in
            make.edges.equalToSuperview().inset(UIEdgeInsets(top: 0, left: defaultMargin, bottom: 0, right: defaultMargin))
        }
        
        contentLabel.snp.makeConstraints { (make) in
            make.edges.equalToSuperview().inset(UIEdgeInsets(top: 8, left: 8 , bottom: 8, right: 8))
        }
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}


